### PR TITLE
figma-linux: init at 0.10.0

### PR DESCRIPTION
**Pull request by O8888**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -482,6 +482,8 @@ with pkgs;
 
   expressvpn = callPackage ../applications/networking/expressvpn { };
 
+  figma = callPackage ../applications/graphics/figma {};
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:485]: It seems like upstream prefers the name `figma-linux` and most Linux packages they mention in their README uses this. That's why I would stick to it as well.

```suggestion
  figma-linux = callPackage ../applications/graphics/figma-linux {};
```

File: pkgs/applications/graphics/figma/default.nix [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{
+  pkgs,
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  enable-csd ? true,
```
Comment (Mic92) [inline pkgs/applications/graphics/figma/default.nix:6]: What does csd stands for?

**Review by dasj19 — COMMENTED** [review]
nix-review says that the package is marked as broken. ídk why it behaves like that. any ideas?

**Comment by dasj19** [comment]
Now nix-review gets stuck at this line:

`[1/0/2 built] building figma-linux-0.10.0 (buildPhase): npm WARN old lockfile }`

I waited for an hour and nothing happened

**Review by dasj19 — APPROVED** [review]
Now it builds with nix-review and the binary works. However I have not tested the functionality behind login.

**Pull request merged**